chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UpdateListingStatusJob.java

```diff
@@ -0,0 +1,163 @@
+
+        Date statusDate = getStatusEffectiveDate(jobContext);
+
+        // This holds all of the Futures, so we can tell when all async processes have ended
+        List<CompletableFuture<Void>> allUpdates = new ArrayList<CompletableFuture<Void>>();
+
+        for (Long cpId : listings) {
+            // This will get listing details, update the listing with the new status, and update listing
+            CompletableFuture<Void> cpFuture = CompletableFuture
+                    .supplyAsync(() -> getListing(cpId))
+                    .thenAccept(cp -> updateListing(cp, certificationStatus, statusDate));
+
+            allUpdates.add(cpFuture);
+        }
+
+        blockUntilAllFuturesComplete(allUpdates);
+
+        LOGGER.info("********* Completed the Update Listing Status job. *********");
+    }
+
+    private CertificationStatus getCertificationStatus(JobExecutionContext context) {
+        @SuppressWarnings("unchecked") Map<String, Object> csMap = (Map<String, Object>) context.getMergedJobDataMap()
+                .get("certificationStatus");
+
+        CertificationStatus cs = new CertificationStatus();
+        cs.setId(Long.parseLong(csMap.get("id").toString()));
+        cs.setName(csMap.get("name").toString());
+        return cs;
+    }
+
+    private void blockUntilAllFuturesComplete(List<CompletableFuture<Void>> allUpdates) {
+        CompletableFuture<Void> combinedFuture = CompletableFuture
+                .allOf(allUpdates.toArray(new CompletableFuture[allUpdates.size()]));
+
+        try {
+            combinedFuture.get();
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
+    private CertificationStatusEvent getCertifiectionStatusEvent(CertificationStatus cs, Date effectiveDate) {
+        CertificationStatusEvent cse = new CertificationStatusEvent();
+        cse.setStatus(cs);
+        cse.setEventDate(effectiveDate.getTime());
+
+        return cse;
+    }
+
+    private CertifiedProductSearchDetails getListing(Long cpId) {
+        try {
+            CertifiedProductSearchDetails cpsd = certifiedProductDetailsManager.getCertifiedProductDetails(cpId);
+            return cpsd;
+        } catch (Exception e) {
+            LOGGER.error(e);
+            return null;
+        }
+    }
+
+    private void updateListing(CertifiedProductSearchDetails cpd, CertificationStatus cs, Date effectiveDate) {
+        cpd.getCertificationEvents().add(getCertifiectionStatusEvent(cs, effectiveDate));
+        ListingUpdateRequest updateRequest = new ListingUpdateRequest();
+        updateRequest.setListing(cpd);
+
+        try {
+            certifiedProductManager.update(
+                    Long.parseLong(updateRequest.getListing().getCertifyingBody().get("id").toString()), updateRequest);
+        } catch (ValidationException e) {
+            LOGGER.error("Error validating {" + cpd.getId() + "}: " + cpd.getChplProductNumber());
```

Comment:
Did you want to add that bit of code we talked about to gather the errors into a list and write that out to CSV and email it? If that needs to be done as an enhancement to this job under the "Retire all 2014 listings" ticket please make sure that gets mentioned in the description. The test cases for that ticket already indicate they are expecting it.

Reply:
Talked about this at stand-up today, and given the desire to get this out to PROD on Monday the decision was made to hold off on emailing stuff until that 2014 ticket

Reply:
We decided to remove that functionality from this tick at this morning's standup.